src/query/src/dist_plan/planner.rs

```diff
@@ -72,66 +71,62 @@ impl ExtensionPlanner for DistExtensionPlanner {
-                let input_plan = merge_scan.input();
-                let optimized_input =
-                    self.optimize_input_logical_plan(session_state, input_plan)?;
-                let input_physical_plan = planner
-                    .create_physical_plan(&optimized_input, session_state)
-                    .await?;
-                let Some(table_name) = self.get_table_name(input_plan)? else {
-                    // no relation found in input plan, going to execute them locally
-                    return Ok(Some(input_physical_plan));
-                };
-
-                let input_schema = input_physical_plan.schema().clone();
-                let input_plan = self.set_table_name(&table_name, input_plan.clone())?;
-                let substrait_plan: Bytes = DFLogicalSubstraitConvertor
-                    .encode(&input_plan)
-                    .context(error::EncodeSubstraitLogicalPlanSnafu)?
-                    .into();
-                let peers = self.get_peers(&table_name).await;
-                match peers {
-                    Ok(peers) => {
-                        let exec = MergeScanExec::new(
-                            table_name,
-                            peers,
-                            substrait_plan,
-                            input_schema,
-                            self.clients.clone(),
-                        );
-
-                        Ok(Some(Arc::new(exec) as _))
-                    }
-                    Err(_) => Ok(Some(input_physical_plan)),
-                }
-            }
-        } else {
-            Ok(None)
+        let Some(merge_scan) = node.as_any().downcast_ref::<MergeScanLogicalPlan>() else {
+            return Ok(None);
+        };
+
+        let input_plan = merge_scan.input();
+        let fallback = |logical_plan| async move {
+            planner
+                .create_physical_plan(logical_plan, session_state)
+                .await
+                .map(Some)
+        };
+
+        if merge_scan.is_placeholder() {
+            // ignore placeholder
+            return fallback(input_plan).await;
         }
+
+        let optimized_input = self.optimize_input_logical_plan(session_state, input_plan)?;
+        let Some(table_name) = Self::extract_full_table_name(input_plan)? else {
+            // no relation found in input plan, going to execute them locally
+            return fallback(&optimized_input).await;
+        };
+
+        let Ok(peers) = self.get_peers(&table_name).await else {
+            // no peers found, going to execute them locally
+            return fallback(&optimized_input).await;
+        };
+
+        // TODO(ruihang): generate different execution plans for different variant merge operation
+        let schema = optimized_input.schema().as_ref().into();
+        let amended_plan = Self::plan_with_full_table_name(input_plan.clone(), &table_name)?;
```

Comment:
cc @waynexia, pass `optimized_input` or `input_plan`?

Reply:
It's intended. The purpose is to leave the original plan to datanode to optimize